### DISCUSSION TITLE
How do I set 'run' to false after finishing demo?

### DISCUSSION THREAD

**Discussion by ajitesh-kuppa-sivakumar** [gilbarbara/react-joyride#791] [Q&A] [answered]
Hi People,

I'm maintaining a boolean react state to determine if the joyride should run or not. When the user travels through joyride and finally clicks on 'Last' button, which will close the joyride. 

But, the problem is, when the user re-visits the same page again, the joyride starts. 

How do I set 'run' to false when the user clicks on 'Last' button (programmatically) ?

Thanks in advance.

**Comment by gilbarbara** [comment, marked as answer]
hey @ajitesh-sivakumar 

You can set a `callback` in the Joyride component and change your state when the tour ends.
https://docs.react-joyride.com/props
